lucene/core/src/java/org/apache/lucene/search/IndexSearcher.java

```diff
@@ -211,6 +215,18 @@ public IndexSearcher(IndexReaderContext context, Executor executor) {
     assert context.isTopLevel: "IndexSearcher's ReaderContext must be topLevel for reader" + context.reader();
     reader = context.reader();
     this.executor = executor;
+    this.sliceExecutionControlPlane = executor == null ? null : getSliceExecutionControlPlane(executor);
+    this.readerContext = context;
+    leafContexts = context.leaves();
+    this.leafSlices = executor == null ? null : slices(leafContexts);
+  }
+
+  // Package private for testing
+  IndexSearcher(IndexReaderContext context, Executor executor, SliceExecutionControlPlane sliceExecutionControlPlane) {
```

Comment:
To quote @jpountz :
"this doesn't need an executor, does it?"

It is needed since executor is a final member of IndexSearcher hence needs to be initialized? We could potentially always set it null since it anyways wont be used but that seemed a bit counter intuitive. WDYT?

Reply:
Is there anything we need to do with the executor that we couldn't do with the sliceExecutionControlPlane?